server/service/orbit_client.go

```diff
@@ -124,14 +137,48 @@ func NewOrbitClient(
 	}
 
 	nodeKeyFilePath := filepath.Join(rootDir, constant.OrbitNodeKeyFileName)
-	return &OrbitClient{
-		nodeKeyFilePath:   nodeKeyFilePath,
-		baseClient:        bc,
-		enrollSecret:      enrollSecret,
-		hostInfo:          orbitHostInfo,
-		enrolled:          false,
-		onGetConfigErrFns: onGetConfigErrFns,
-	}, nil
+	oc := &OrbitClient{
+		nodeKeyFilePath:            nodeKeyFilePath,
+		baseClient:                 bc,
+		enrollSecret:               enrollSecret,
+		hostInfo:                   orbitHostInfo,
+		enrolled:                   false,
+		onGetConfigErrFns:          onGetConfigErrFns,
+		lastIdleConnectionsCleanup: time.Now(),
+	}
+
+	return oc, nil
+}
+
+// closeIdleConnections attempts to close idle connections from the pool
+// every 10 minutes.
+//
+// Some load balancers (e.g. AWS ELB) have a maximum lifetime for a connection
+// (no matter if the connection is active or not) and will forcefully close the
+// connection causing errors in the client (e.g. https://github.com/fleetdm/fleet/issues/18783).
+// To prevent these errors, we will attempt to cleanup idle connections every 10
+// minutes to not let these connection grow too old. (AWS ELB's default value for maximum
+// lifetime of a connection is 3600 seconds.)
+func (oc *OrbitClient) closeIdleConnections() {
+	oc.lastIdleConnectionsCleanupMu.Lock()
+	defer oc.lastIdleConnectionsCleanupMu.Unlock()
+
+	if time.Since(oc.lastIdleConnectionsCleanup) < 10*time.Minute {
```

Comment:
If the connection is closed, we must do a TLS handshake again, right?
So why not close less often, like every 55 minutes?

Reply:
Yes, I think 55m is a good idea. (It was 10m because I initially ran this on a for loop and there was the chance of connections being not closed because they were in use)